apps/design/frontend/src/ballot_screen.tsx

```diff
@@ -314,6 +314,30 @@ export function BallotScreen(): JSX.Element | null {
 
             if (ballotResult.isErr()) {
               const err = ballotResult.err();
+              if (err.error === 'missingRequiredField') {
+                return (
+                  <Row
+                    style={{
+                      justifyContent: 'center',
+                      alignItems: 'center',
+                      height: '100%',
+                    }}
+                  >
+                    <Callout color="danger" icon="Danger">
+                      <span>
+                        The election is missing the following required field:
+                        &quot;{err.field}&quot;. Update in{' '}
+                        <Link
+                          to={routes.election(electionId).electionInfo.path}
+                        >
+                          Election Info
+                        </Link>
```

Comment:
I like the inclusion of the link to fix the issue!

I think it's a little early to be abstracting a pattern for missing required fields, since they aren't all updated in the Election Info screen and may need different instructions.  Maybe we just make a single specific error type and user-facing message for missing signatures to start and see if we need a more general solution later? Then the message could be worded a little more directly, e.g. "Missing signature. Upload a signature in Election Info."

Reply:
Thanks for that feedback, I was feeling uncertain on which path to take, but I do agree with what youv'e said. I'll go with a single specific error type!